packages/flutter/lib/src/painting/image_stream.dart

```diff
@@ -155,11 +213,15 @@ class ImageStream extends Diagnosticable {
   /// discouraged because [removeListener] will remove the first instance that
   /// was added, even if it was added with a different `onError` than the
   /// intended paired `addListener` call.
-  void addListener(ImageListener listener, { ImageErrorListener onError }) {
+  void addListener(
```

Comment:
this api is already broken and this makes it more broken. i don't think we should go down this road.

See the docs for removeListener, for example. We'd have to update those if we went this path, and I don't think I could do that without shame.